### PR TITLE
Implementing Diagnostics HTTP API.

### PR DESCRIPTION
**Pull request by nivancevic**

### CODE REVIEW

File: cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/Diagnostics.java [hunk 1]
```diff
@@ -87,15 +92,24 @@ private Configuration loadConfiguration() {
...
         final MBeanServer server = ManagementFactory.getPlatformMBeanServer();
+        final DiagnosticsMXBean mbean = new DiagnosticsMXBeanImpl(config, this);
         try {
             mxbeanName = new ObjectName(
                     DiagnosticsMXBean.class.getPackage() + ":type=" + DiagnosticsMXBean.class.getSimpleName());
-            final DiagnosticsMXBean mbean = new DiagnosticsMXBeanImpl(config, this);
             server.registerMBean(mbean, mxbeanName);
         } catch (MalformedObjectNameException | InstanceAlreadyExistsException | MBeanRegistrationException |
                 NotCompliantMBeanException e) {
             logger.error("Unable to register DiagnosticsMBean", e);
```
Comment (mgobec) [inline cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/Diagnostics.java:102]: If this fails we will continue with api init.

File: cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/Diagnostics.java [hunk 2]
```diff
@@ -87,15 +92,24 @@ private Configuration loadConfiguration() {
...
         }
+        if (config.httpApiEnabled && httpApi == null) {
+            logger.info("Starting Diagnostics HTTP API at {}:{}", config.httpApiHost, config.httpApiPort);
+            httpApi = new HttpHandler(config, mbean);
+            try {
```
Comment (mgobec) [inline cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/Diagnostics.java:107]: Can this be nested in top try/catch?

File: cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/api/DiagnosticsMXBean.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package io.smartcat.cassandra.diagnostics.jmx;
+package io.smartcat.cassandra.diagnostics.api;
```
Comment (mgobec) [inline cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/api/DiagnosticsMXBean.java]: Why package rename?

Comment (nivancevic) [reply cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/api/DiagnosticsMXBean.java]: There is nothing related to JMX in this package actually. And it is API.

File: cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/Diagnostics.java (hunk 1 above)
Comment (nivancevic) [reply cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/Diagnostics.java:102]: Yes. With error message logged.

**Pull request merged**